system-addon/lib/TelemetryFeed.jsm

```diff
@@ -118,6 +120,14 @@ this.TelemetryFeed = class TelemetryFeed {
     return this.pingCentre;
   }
 
+  /**
+   * Lazily initialize UTEventReporting to send pings
+   */
+  get utEvents() {
+    Object.defineProperty(this, "utEvents", {value: new UTEventReporting()});
```

Comment:
Does this need a guard here to check if `utEvents` already exists?

Reply:
I think by default when you use `Object.defineProperty()` and set its `value` it can no longer have a getter or setter. So next time `this.utEvents` is called, it will access the property value directly and will not try to create a `new UTEventReporting()`. We do the same thing for `PingCentre` here:

https://github.com/mozilla/activity-stream/blob/master/system-addon/lib/TelemetryFeed.jsm#L110